third_party/move/move-vm/runtime/src/execution_tracing/loggers.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Interfaces and implementations for [Trace] collection.
+
+use crate::{
+    execution_tracing::{
+        trace::{CondBrTrace, DynamicCall},
+        Trace,
+    },
+    LoadedFunction,
+};
+use move_core_types::function::ClosureMask;
+
+/// Interface for recording the trace at runtime. It is sufficient to record branch decisions as
+/// well as dynamic function calls originating from closures.
+pub trait TraceLogger {
```

Comment:
I slightly prefer TraceRecorder, APIs are record_*, and also logging is a bit ambiguous / also typically not compulsory but good to have, but here we must record everything.

Reply:
Renamed